htdocs/core/lib/security.lib.php

```diff
@@ -144,6 +144,10 @@ function dolEncrypt($chain, $key = '', $ciphering = 'AES-256-CTR', $forceseed =
 	$newchain = $chain;
 
 	if (function_exists('openssl_encrypt') && empty($dolibarr_disable_dolcrypt_for_debug)) {
+                if (empty($key)) {
+                        return 'Error dolEncrypt encrypt key is empty';
```

Comment:
Instead of return an error string, i would make
return $chain;
So encryption is disabled with nobrisknto corrupt data with an error string when we save the result in database.